pkg/components/validate_headers_test.go

```diff
@@ -14,63 +14,66 @@ func Test_incomingMatchesAllowed(t *testing.T) {
 		name           string
 		allowedHeader  map[string][]string
 		incomingHeader map[string][]string
-		wantResult     bool
 		wantErr        bool
 	}{
 		{
 			name:           "multiple incoming header values with multiple allowed values",
 			allowedHeader:  defaultAllowedHeaderAndValues,
 			incomingHeader: map[string][]string{"Ldap-Groups": {"sre", "devs"}},
-			wantResult:     true,
 			wantErr:        false,
 		},
 		{
 			name:           "multiple incoming header values with a single allowed value",
 			allowedHeader:  map[string][]string{"Ldap-Groups": {"devs"}},
 			incomingHeader: map[string][]string{"Ldap-Groups": {"sre", "devs"}},
-			wantResult:     true,
+			wantErr:        false,
+		},
+		{
+			name:           "multiple incoming header values separated by a comma with a single allowed value",
+			allowedHeader:  map[string][]string{"Ldap-Groups": {"devs"}},
+			incomingHeader: map[string][]string{"Ldap-Groups": {"sre,ops", "security,devs"}},
+			wantErr:        false,
+		},
+		{
+			name:           "single incoming header value separated by a comma with a single allowed value",
+			allowedHeader:  map[string][]string{"Ldap-Groups": {"ops"}},
+			incomingHeader: map[string][]string{"Ldap-Groups": {"security,ops"}},
 			wantErr:        false,
 		},
 		{
 			name:           "single incoming header value with multiple allowed values",
 			allowedHeader:  defaultAllowedHeaderAndValues,
 			incomingHeader: map[string][]string{"Ldap-Groups": {"devs"}},
-			wantResult:     true,
```

Comment:
didn't need to check for this anymore now the method no longer returns a boolean and only returns an error if no match is found